src/main/kotlin/org/rust/lang/core/grammar/rust.bnf

```diff
@@ -887,8 +882,9 @@ cast_expr ::= restricted_expr AS type
 
 unary_expr ::= ('-' | '*' | '!' | ('&' | '&&') MUT?) restricted_expr
 
-lambda_expr ::= MOVE? ('||' | '|' [ <<comma_separated_list lambda_param>> ] '|') ret_type? any_expr
-lambda_param ::= pat type_ascription?
+lambda_expr   ::= MOVE? lambda_params ret_type? any_expr
+lambda_params ::= '||' | '|' [ <<comma_separated_list lambda_param>> ] '|' { elementType=params }
```

Comment:
What do you think about removing `||` token from lexer, and use `collapse` wherever it's needed?